frontend/amundsen_application/static/js/features/ColumnList/ColumnDetailsPanel/index.tsx

```diff
@@ -0,0 +1,131 @@
+// Copyright Contributors to the Amundsen project.
+// SPDX-License-Identifier: Apache-2.0
+
+import * as React from 'react';
+import EditableSection from 'components/EditableSection';
+import BadgeList from 'features/BadgeList';
+import ColumnDescEditableText from 'features/ColumnList/ColumnDescEditableText';
+import ColumnLineage from 'features/ColumnList/ColumnLineage';
+import ColumnStats from 'features/ColumnList/ColumnStats';
+import ExpandableUniqueValues from 'features/ExpandableUniqueValues';
+import { getMaxLength, isColumnListLineageEnabled } from 'config/config-utils';
+import { getColumnLink } from 'utils/navigationUtils';
+import { filterOutUniqueValues, getUniqueValues } from 'utils/stats';
+import { FormattedDataType } from '..';
+import {
+  COPY_COL_LINK_LABEL,
+  COPY_COL_NAME_LABEL,
+  EDITABLE_SECTION_TITLE,
+  CLOSE_LABEL,
+} from './constants';
+
+export interface ColumnDetailsPanelProps {
+  columnDetails: FormattedDataType;
+  togglePanel: (columnDetails: FormattedDataType, event: any) => void;
+}
+
+const shouldRenderDescription = (columnDetails: FormattedDataType) => {
+  const { content, editText, editUrl, isEditable } = columnDetails;
+  if (content.description) {
+    return true;
+  }
+  if (!editText && !editUrl && !isEditable) {
+    return false;
+  }
+  return true;
+};
+
+const ColumnDetailsPanel: React.FC<ColumnDetailsPanelProps> = ({
+  columnDetails,
+  togglePanel,
+}: ColumnDetailsPanelProps) => {
+  const {
+    content,
+    stats,
+    editText,
+    editUrl,
+    col_index,
+    name,
+    tableParams,
+    isEditable,
+    badges,
+  } = columnDetails;
+
+  const normalStats = stats && filterOutUniqueValues(stats);
+  const uniqueValueStats = stats && getUniqueValues(stats);
+
+  return (
+    <aside className="right-panel">
+      <div className="panel-header">
+        <div className="panel-title">{name}</div>
+        <button
+          type="button"
+          className="btn btn-close"
+          onClick={togglePanel.bind(null, null)}
+        >
+          <span className="sr-only">{CLOSE_LABEL}</span>
+        </button>
+      </div>
+      <div className="buttons-row">
+        <button
+          className="btn btn-default column-button"
+          id="copy-col-name"
+          type="button"
+          onClick={() => {
+            navigator.clipboard.writeText(name);
+          }}
+        >
+          {COPY_COL_NAME_LABEL}
+        </button>
+        <button
+          className="btn btn-default"
+          id="copy-col-link"
+          type="button"
+          onClick={() => {
+            const link = getColumnLink(tableParams, name);
+            navigator.clipboard.writeText(link);
```

Comment:
Removing the temporal would probably work here better:
```suggestion
            navigator.clipboard.writeText(getColumnLink(tableParams, name));
```

Reply:
That said, I prefer to extract a 'handleCopyButtonClick'